lib/reek/smells/smell_warning.rb

```diff
@@ -53,12 +55,17 @@ def matches_smell_type?(klass)
         smell_classes.include?(klass.to_s)
       end
 
-      def matches_smell_details?(other_smell_details = {})
-        common_parameters_equal?(other_smell_details)
+      def matches_attributes?(other_attributes = {})
+        check_attributes_comparability(other_attributes)
+        return false unless common_parameters_equal?(other_attributes[:parameters])
+        other_attributes.delete(:parameters)
```

Comment:
Why not invert this logic, so you take out the values for the keys message, lines, context and source, and assume the rest are parameters? This avoids the extra indirection for the parameters.